Client/FeatureFlags/FlaggableFeature.swift

```diff
@@ -203,3 +203,15 @@ extension FlaggableFeature {
         }
     }
 }
+
+func soakNimbus<T>(featureId: String, iterations: Int = 100_000, in file: String = #file, at lineno: Int = #line, creator: @escaping (Variables) -> T) {
+    print("Start soaking \(featureId)")
+    let getNimbus = { Experiments.shared }
+    for _ in 1..<iterations {
+        let holder = FeatureHolder<T>(getNimbus, featureId, creator)
+        _ = holder.value()
```

Comment:
I can reproduce the crash reliably and only made the following change:
```suggestion
      let _ = FxNimbus.shared.features.homescreen.value()
```

Using an iPhone 13 simulator, I am always seeing the crash after about 1 minute. There are two flavours of this crash. The more commonly occurring SIGABRT (non-allocated pointer ... being freed), and EXC_BAD_ACCESS (EXC_I386_GPFLT) which we also see in Sentry.

The fact it happens at roughly the same time on every run suggest it's triggered by memory pressure possibly.

Reply:
The fact that this only reproduces going through `FxNimbus.shared.features.homescreen` is a good hint too. The `.value` call doesn't crash without my change when going through a new `FeatureHolder` instance every time.

/cc @jhugman @nbhasin2

Reply:
OK, after more investigation it's clear now why the crash happens at roughly the same amount of time in this testing branch:

In [FirefoxHomeViewController](https://github.com/mozilla-mobile/firefox-ios/blob/e5b7ca8dd365e74b86e22019e1ce05563a185dab/Client/Frontend/Home/FirefoxHomeViewController.swift#L634), we schedule an async update of the view model on a separate thread, which ends up causing a concurrent call on `FeatureHolder.value`, causing the crash:

<img width="485" alt="Screen Shot 2022-04-18 at 5 41 38 PM" src="https://user-images.githubusercontent.com/472523/163884006-f89cde12-de59-416c-9f3c-b649f553391d.png">



Access to `FeatureHolder.value` is not currently thread-safe. 

This also explains why this crash is not reproducible using a fresh new (local) `FeatureHolder` instance every time, but only when using the shared global instance.

In reality this concurrent overlapping call will be less likely than in this test branch, but that explains why it's hard to reproduce manually.

Reply:
Confirmed that I am no longer seeing the crash after commenting out https://github.com/mozilla-mobile/firefox-ios/blob/e5b7ca8dd365e74b86e22019e1ce05563a185dab/Client/Frontend/Home/FirefoxHomeViewController.swift#L634.

I think this confirms that we either need to make `FeatureHolder.value` thread-safe in Nimbus, or prevent concurrent calls to it in the application.